docs/network/calico-network.md

```diff
@@ -0,0 +1,196 @@
+
+1.  Container to Container (Which collectively provides a service) communications
+1.  Pod to Pod Communication
+1.  Pod to Service Communication
+1.  External to Service Communication
+
+## Pod to pod communication
+
+By default docker creates a virtual bridge called “docker0” on the host machine and it assigns a private network range to it. For each container that is created, a virtual Ethernet device is attached to this bridge, which is then mapped to eth0 inside the container, with an IP from the network range. This happens for each host that is running docker. There is no coordination between these hosts therefore the network ranges might collide.
+
+Because of this, containers will only able to communicate with containers that are connected to
+the same virtual bridge. To communicate with other containers on other hosts, they must rely on port mapping. This means that you need to assign a port on the host machine to each container, and then forward all traffic on that port to that container.
+
+Since local IP address of the application is translated to the host IP address and port on the host machine, Kubernetes assumes that all the nodes can communicate with each other without NAT and that the IP address that a container sees for itself is the same that the other containers see for it. Aside from being simpler, it also enables applications to be ported rather easily from virtual machines to containers, since they do not have to change the way they work in terms of network.
+
+Calico is one of the many different networking options that offer these capabilities for Kubernetes.
+
+## Calico
+
+Calico is designed to simplify, scale, and secure cloud networks. The open source framework
+enables Kubernetes networking and network policy for clusters across the cloud. Within the Kubernetes ecosystem, Calico is starting to emerge as one of the most popularly used network frameworks or plug-ins, with many enterprises using it at scale. 
+
+Calico uses a pure IP networking fabric to deliver high performance Kubernetes networking, and its policy engine enforces developer intent for high-level network policy management. Calico provides Layer 3 networking capabilities and associates a virtual router with each node. It enables host to host and pod to pod networking. Calico allows establishment of zone boundaries through BGP or encapsulation through IP on IP or VXLAN methods.
+
+## Integration between Kubernetes and Calico
+
+Calico integrates with Kubernetes through CNI plug-in built on a fully distributed, layer 3 architecture hence it scales smoothly from a single laptop to large enterprise. It relies on an IP layer and it is relatively easy to debug with existing tools.
+
+### Configure the network with Calico
+
+First, bring up a Kubernetes cluster with Calico using the following commands:
+
+    > kubeadm init --pod-network-cidr=192.168.0.0/16
```

Comment:
If there is more than one k8s cluster behind the ADC they will require different subnet ranges but the same ASN to function.